gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java

```diff
@@ -213,6 +217,57 @@ private SolrQuery buildTimeSeriesChunkQuery(JsonObject params) {
         query.setRows(params.getInteger(MAX_TOTAL_CHUNKS_TO_RETRIEVE_REQUEST_FIELD, 50000));
         return query;
     }
+    private SolrQuery buildAnnotationQuery(JsonObject params) {
+        StringBuilder queryBuilder = new StringBuilder();
+        Long from = params.getLong(FROM_REQUEST_FIELD, null);
+        Long to = params.getLong(TO_REQUEST_FIELD, null);
+        if (to == null && from != null) {
+            LOGGER.trace("requesting annotation with from time {}", from);
+            queryBuilder.append(TIME_REQUEST_FIELD).append(":[").append(from).append(" TO ").append("*]");
+        } else if (from == null && to != null) {
+            LOGGER.trace("requesting annotation with to time {}", to);
+            queryBuilder.append(TIME_REQUEST_FIELD).append(":[*").append(" TO ").append(to).append("]");
+        } else if (from != null) {
+            LOGGER.trace("requesting annotation with time from {} to time {}", from, to);
+            queryBuilder.append(TIME_REQUEST_FIELD).append(":[").append(from).append(" TO ").append(to).append("]");
+        } else {
+            LOGGER.trace("requesting annotation with all times existing");
+            queryBuilder.append("*:*");
+        }
+        //FILTER
+        List<String> tags = null;
+        if (params.getJsonArray(TAGS_REQUEST_FIELD, null) != null)
+            tags = params.getJsonArray(TAGS_REQUEST_FIELD).getList();
+        StringBuilder stringQuery = new StringBuilder();
+        String operator = "";
+        SolrQuery query = new SolrQuery();
+        if (params.getString(TYPE_REQUEST_FIELD, "all").equals("tags")) {
+            queryBuilder.append(" && ");
+            if (!params.getBoolean(MATCH_ANY_REQUEST_FIELD, true)) {
+                operator = " AND ";
+            } else {
+                operator = " OR ";
+            }
+            for (String tag : tags.subList(0,tags.size()-1)) {
+                stringQuery.append(tag).append(operator);
+            }
+            stringQuery.append(tags.get(tags.size()-1));
+            queryBuilder.append(TAGS_TO_FILTER_ON_REQUEST_FIELD).append(":").append("(").append(stringQuery.toString()).append(")");
+        }
+        if (queryBuilder.length() != 0 ) {
```

Comment:
even if queryBuilder length is 0 you should put things like number of rows to return....